scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/EcdhPskClientKeyExchange.java

```diff
@@ -0,0 +1,146 @@
+/*******************************************************************************
+ * Copyright 2018 University of Rostock, Institute of Applied Microelectronics and Computer Engineering
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Vikram (University of Rostock)- Initial creation, adapted from ECDHClientKeyExchange
+ ******************************************************************************/
+package org.eclipse.californium.scandium.dtls;
+
+import static org.eclipse.californium.elements.util.StandardCharsets.UTF_8;
+
+import java.net.InetSocketAddress;
+import java.security.PublicKey;
+import java.security.interfaces.ECPublicKey;
+import java.security.spec.ECParameterSpec;
+import java.security.spec.ECPoint;
+import java.util.Arrays;
+
+import org.eclipse.californium.elements.util.DatagramReader;
+import org.eclipse.californium.elements.util.DatagramWriter;
+import org.eclipse.californium.elements.util.StringUtil;
+import org.eclipse.californium.scandium.dtls.cipher.ECDHECryptography;
+import org.eclipse.californium.scandium.util.ByteArrayUtils;
+
+/**
+ * class EcdhPskClientKeyExchange. <br>
+ * This class EcdhPskClientKeyExchange represents the {@link ClientKeyExchange} in ECDHE_PSK mode. See <a
+ * href="https://tools.ietf.org/html/rfc5489#section-2">RFC 5489</a> for details.
+ * @author Vikram
+ *
+ */
+public final class EcdhPskClientKeyExchange extends ClientKeyExchange{
+
+
+	protected static final int LENGTH_BITS = 8; // opaque point <1..2^8-1>
+	private static final int IDENTITY_LENGTH_BITS = 16; // opaque <0..2^16-1>;
+
+	/**
+	 *See <a href="https://tools.ietf.org/html/rfc5489#section-2">RFC 5489</a>.
+	 */
+	private final byte[] identityEncoded;
+	/**
+	 *See <a href="https://tools.ietf.org/html/rfc5489#section-2">RFC 5489</a>.
+	 */
+	private final String identity;
+	private final byte[] pointEncoded;
+	
+	/**
+	 * Constructor {@link EcdhPskClientKeyExchange}. The {@link EcdhPskClientKeyExchange} message include 
+	 * an psk identity as clear text for server to identify which psk key is used, 
+	 * and the ephemeral public key of the client.  
+	 * @param hint - PSK identity as clear text
+	 * @param clientPublicKey - ephemeral public key of client
+	 * @param peerAddress
+	 */
+	public EcdhPskClientKeyExchange(String hint, PublicKey clientPublicKey, InetSocketAddress peerAddress) {
```

Comment:
null checks?

Reply:
Do I have to introduce Assert.assertNotNull(clientPublicKey);  for this?

Reply:
See DataParser.parseMessage

https://github.com/eclipse/californium/blob/2.0.x/californium-core/src/main/java/org/eclipse/californium/core/network/serialization/DataParser.java

And please don't forget to document the throws NullPointerException.